### DISCUSSION TITLE
How to extend the existing debug experience

### DISCUSSION THREAD

**Discussion by mrlacey** [VsixCommunity/Community.VisualStudio.Toolkit#297] [Q&A]
[Official documentation](https://docs.microsoft.com/en-us/visualstudio/extensibility/debugger/visual-studio-debugger-extensibility?view=vs-2022) only shows how to add debugger support for entirely new languages. I can't find anything about extending the debug experience for a language that already has a debugger.

In my case, I often have file and directory names inside variables while I'm debugging C# code and I want to have an easy way of opening those files & directories.

Ideally, I want to add entries to this menu and have access to the value so that I can open it. (Screenshot taken at a random point and not representative of my actual use case.) 

![image](https://user-images.githubusercontent.com/189547/152821545-6b86d375-b762-4c34-8712-c4c7fd8ac3a4.png)

Currently, I have to copy the value, open an app (notepad) and then open a file and paste in the value for the file name. Or I have to open Explorer, switch to the address bar, and then paste in the value I've copied.
It might not sound like an onerous process but do it a few dozen times each day and you'll think otherwise.

Can anyone help?

**Comment by moon6969** [comment]
I want my Extension to eventually be able to snoop on variable values during debug.

When I looked into it some time back I noted this as a starting point...
[EnvDTE.Debugger.CurrentStackFrame](https://docs.microsoft.com/en-us/dotnet/api/envdte.debugger.currentstackframe)
[EnvDTE.StackFrame.Locals](https://docs.microsoft.com/en-us/dotnet/api/envdte.stackframe.locals)

I'm not sure if there is a more modern way to do it witout DTE.
